File: java/minimum_size_subarray_sum/minimum_size_subarray_sum.java
package minimum_size_subarray_sum;

class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        double min_length = Double.POSITIVE_INFINITY;
        int summ = 0;
        int l = 0;

        for (int r = 0; r < nums.length; r++) {
            summ += nums[r];
            while (summ >= target) {
                min_length = Math.min(min_length, r - l + 1);
                summ -= nums[l];
                l += 1;
            }
        }
        return (int) (min_length < Double.POSITIVE_INFINITY ? min_length : 0.0);
    }
}

File: c++/minimum_size_subarray_sum/minimum_size_subarray_sum.cpp
#include <vector>
#include <limits>
#include <algorithm>
#include <iostream>

using namespace std;
class Solution
{
public:
    int minSubArrayLen(int target, vector<int> &nums)
    {
        int min_length = std::numeric_limits<int>::max();
        int summ = 0;
        int l = 0;
        for (int r = 0; r < nums.size(); r++)
        {
            summ += nums[r];
            while (summ >= target)
            {
                min_length = std::min(min_length, r - l + 1);
                summ -= nums[l];
                l += 1;
            }
        }
        return (min_length < std::numeric_limits<int>::max()) ? min_length : 0;
    }
};